.github/copilot-instructions.md

```diff
@@ -0,0 +1,142 @@
+- **K8s**: Operator testing (`k8s-eks`, `k8s-gke`, `k8s-local-kind`)
+
+## Essential Workflows
+
+### Running Tests Locally
+```bash
+# Docker backend (recommended for development)
+export SCT_SCYLLA_VERSION=5.2.1
+hydra run-test longevity_test.LongevityTest.test_custom_time \
+  --backend docker \
+  --config test-cases/PR-provision-test-docker.yaml
+
+# AWS backend (requires credentials)
+export SCT_ENABLE_ARGUS=false  # Disable reporting during testing
+hydra run-test longevity_test.LongevityTest.test_custom_time \
+  --backend aws \
+  --config test-cases/PR-provision-test.yaml \
+  --config configurations/network_config/test_communication_public.yaml
+```
+
+### Configuration System
+- **Test configs**: `test-cases/` directory contains YAML configurations
+- **Shared configs**: `configurations/` for reusable settings (network, nemesis, etc.)
+- **Multiple configs**: Chain with multiple `-c` flags - later configs override earlier ones
+- **Environment variables**: Prefix with `SCT_` (e.g., `SCT_SCYLLA_VERSION=5.2.1`)
+
+### Nemesis (Chaos Engineering)
+- **Nemesis classes**: Defined in `sdcm/nemesis.py` - inherit from `Nemesis` base class
+- **Generation tools**: Use `hydra create-nemesis-pipelines` to generate Jenkins pipelines
+- **Docker limitations**: Some nemesis types skip on Docker backend (no hard reboot support)
+
+## Project-Specific Patterns
+
+### Test Configuration Inheritance
+```yaml
+# Base config
+test_duration: 180
+n_db_nodes: 6
+nemesis_class_name: 'SisyphusMonkey'
+
+# Override in environment or additional configs
+stress_cmd: ["cassandra-stress write ..."]
+```
+
+### Cluster Access Patterns
+```python
+# Database connections
+with self.db_cluster.cql_connection_patient(node) as session:
+    session.execute("SELECT * FROM keyspace.table")
+
+# Node operations
+for node in self.db_cluster.nodes:
+    node.run_nodetool("repair")
+    node.remoter.run("systemctl restart scylla-server")
+```
+
+### Multi-Tenant Testing
+- Access via `self.db_clusters_multitenant` for multiple clusters
+- Common pattern for cross-DC and multi-region testing
+- Supports parallel cluster operations with `ParallelObject`
+
+### Logging and Results
+- **Test logs**: `~/sct-results/latest/` contains all test artifacts
+- **Test ID**: Generated UUID, access via `self.test_config.test_id()`
+- **Event system**: Use `InfoEvent`, `ErrorEvent` for structured logging
+
+## Development Guidelines
+
+### Local Environment Setup
+```bash
+# Use uv + direnv for modern Python management (see docs/setup-uv-direnv.md)
+# Or traditional setup:
+sudo ./install-hydra.sh
+hydra bash  # Enter containerized environment
+```
+
+### Test Development Pattern
+1. Create test class inheriting from `ClusterTester`
+2. Define test methods with `test_` prefix
+3. Use existing config templates from `test-cases/`
+4. Test locally with Docker backend first
+5. Validate on AWS/GCE for final testing
```

Comment:
I would add section for Nemesis Development pattern, which is more commonly used